src/session/src/context.rs

```diff
@@ -173,6 +191,15 @@ impl QueryContext {
         }
     }
 
+    pub fn update_configuration_parameter(&self, session: &SessionRef) {
+        self.configuration_parameter
+            .clone()
+            .into_iter()
+            .for_each(|(key, value)| {
+                session.configuration_variables.insert(key, value);
```

Comment:
I see dashmap has implemented "[`extend`](https://docs.rs/dashmap/latest/dashmap/struct.DashMap.html#impl-Extend%3C(K,+V)%3E-for-DashMap%3CK,+V,+S%3E)", maybe you can use that?